packages/mui-material/src/Menu/Menu.js

```diff
@@ -164,24 +170,26 @@ const Menu = React.forwardRef(function Menu(inProps, ref) {
         horizontal: isRtl ? 'right' : 'left',
       }}
       transformOrigin={isRtl ? RTL_ORIGIN : LTR_ORIGIN}
-      slots={{ paper: MenuPaper }}
+      slots={{
+        paper: PaperSlot,
+        root: slots?.root,
+      }}
       slotProps={{
+        root: {
+          ...resolveComponentProps(slotProps?.root, ownerState),
```

Comment:
Let's use `useSlotProps` instead of `resolveComponentProps`. The `useSlotProps` hook is meant to handle what we're doing with `resolveComponentProps` manually here (merge classNames, ownerState, external slot props), and it actually [uses `resolveComponentProps`](https://github.com/mui/material-ui/blob/master/packages/mui-base/src/utils/useSlotProps.ts#L87) under the hood.

So the change would look something like this (I haven't tested it so it might have some errors):

```js
// ...

const PaperSlot = slots.paper ?? MenuPaper;
const paperExternalSlotProps = slotProps.paper ?? PaperProps;

const rootSlotProps = useSlotProps({
    elementType: slots.root,
    externalSlotProps: slotProps.root,
    ownerState,
    className: [classes.root. className],
})

const paperSlotProps = useSlotProps({
    elementType: PaperSlot,
    externalSlotProps: paperExternalSlotProps,
    ownerState,
    className: classes.paper, // no need to add paperExpernalSlotProps.className here, useSlotProps will merge it
})

// ...
// inside the MenuRoot props:

slotProps={{
    root: rootSlotProps,
    paper: paperSlotProps,
}}

// ...

```